docs/UserGuide.md

```diff
@@ -3,436 +3,713 @@ layout: page
 
-* Download Java [here](https://www.oracle.com/java/technologies/javase/jdk17-archive-downloads.html). Follow the instructions on that page to install Java.
+4. [Understanding Commands in AgentAssist](#4-understanding-commands-in-agentassist)
+   <details>
+   <summary>Subsections</summary>
 
-### 2. Download the AgentAssist application
+   4.1 [Command Structure Overview](#41-command-structure-overview)  
+   4.2 [Commands](#42-commands)  
+   4.3 [Flags](#43-flags)  
+   4.4 [Arguments](#44-arguments)  
+   4.5 [Using Commands](#45-using-commands)
 
-Go to this [link](https://github.com/AY2425S1-CS2103T-T14-4/tp/releases) and download the latest version of the `.jar` file.
+   </details>
 
-### 3. Choose a Folder
+5. [Commands](#5-commands)
+   <details>
+   <summary>Subsections</summary>
 
-Find or create a folder on your computer where you want to store the AgentAssist information. <br>
-Move the `.jar` file you just downloaded into this folder.
+   5.1 [How to Read Commands](#51-how-to-read-commands)  
+   5.2 [Data Modification Commands](#52-data-modification-commands)  
+   5.3 [Data Filtering Commands](#53-data-filtering-commands)  
+   5.4 [General Commands](#54-general-commands)  
+   5.5 [Saving Data](#55-saving-data)  
+   5.6 [Modifying the Data File](#56-modifying-the-data-file)
 
-### 4. Open a Command Terminal
+   </details>
+   
+6. [FAQ](#6-faq)
+7. [Known Issues](#7-known-issues)
+8. [Command Summary](#8-command-summary)
 
-Now, open a command terminal:
 
-* On Windows, press the **Windows Key** and type **`Command Prompt`** in the search bar. Click on the **Command Prompt** application to open it.
-* On macOS, press **Cmd + Space**, type **`Terminal`** into the search bar, and press **Enter**.
-* On Linux, open your Terminal application.
+# 1. Introduction
+## 1.1 What is AgentAssist?
 
-### 5. Run the applicaton
+AgentAssist is the **definitive desktop tool for credit card sales agents**. Merging the swift efficiency of a Command Line Interface (CLI) with the intuitive accessibility of a Graphical User Interface (GUI), this application lets you manage contact databases, track sales progress, and execute transactions with unprecedented speed. 
 
-Navigate your terminal to the folder where you saved the AgentAssist application:
+**Overview of Key Features:**
+* **Contact Management**: 
+  * Manage your client details easily. Add, edit, and delete contacts to keep all your client information in one accessible place.
+* **Keyboard-only Navigation**: 
+  * Navigate through the application entirely via keyboard shortcuts, improving workflow efficiency.
+* **Multi-Level Filtering**: 
+  * Filter your data by multiple criteria to find exactly who you’re looking for.
+* **Auto-Save**: 
+  * Automatically saves your work as you go, ensuring data is updated without manual intervention.
+* **Effortless Data Import & Export**: 
+  * Import or export client and sales data in compatible formats for backups or use in other systems.
```

Comment:
Nice labelling, but perhaps `Keyboard-only Navigation` is misleading and can be replaced with `Keyboard-Centric Navigation`, as they can still use the mouse for instance to close the app and to scroll.

Reply:
Yup, I agree with this, `keyboard-centric` does fit this scenario better.